src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java

```diff
@@ -82,6 +82,8 @@
   // TODO(ulfjack): Throw an exception if it's not a file.
   public abstract long getSize();
 
+  public abstract String getActionId();
```

Comment:
Consider moving the default implementation (`return "";`) here.

Reply:
Done and commented.